### ISSUE TITLE
Trusted Publishing with reusable workflow

### ISSUE THREAD

**Issue by segiddins** [rubygems/rubygems.org#4294]
Hi. Let me ask about Trusted Publishing here since I can't find where to ask.
(Please tell me a different place if my question is inappropriate 🙏🏼 )

First, my try in the suggested way by the [RubyGems document](https://guides.rubygems.org/trusted-publishing/releasing-gems/) was successful! 🎉 
(For details, see my repo's [Actions log](https://github.com/ybiquitous/easytest/actions/runs/7210080404/job/19642449449#step:4:1))

---

However, my try through a reusable workflow failed, unfortunately. I'm not sure about the failure cause, but `rubygems/configure-rubygems-credentials` failed with the following error:

```
No trusted publisher configured for this workflow found on https://rubygems.org/ for audience rubygems.org
```

See also the [failure Actions log](https://github.com/ybiquitous/easytest/actions/runs/7209639798/job/19641317427#step:4:18).

Here's a part of my Actions workflow settings:

```yaml
# .github/workflows/release.yml in the gem repository
jobs:
  release:
    uses: ybiquitous/.github/.github/workflows/ruby-release-reusable.yml@new-ruby-release
```

```yaml
# The sharable workflow in a different repository
jobs:
  release:
    runs-on: ubuntu-latest
    permissions:
      contents: write
      id-token: write
    timeout-minutes: 10
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Set up Ruby ${{ inputs.ruby-version }}
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ inputs.ruby-version }}
          bundler-cache: true
      - name: Publish gem to RubyGems.org
        uses: rubygems/release-gem@v1
```

For the sharable workflow code, see <https://github.com/ybiquitous/.github/pull/31/files>.

In short, **is Trusted Publishing unsupported in a sharable workflow?**

---

Last, this feature is fantastic! Thanks a lot. 👏🏼

_Originally posted by @ybiquitous in https://github.com/rubygems/rubygems.org/issues/4285#issuecomment-1855990785_

**Comment by segiddins** [comment]
https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/using-openid-connect-with-reusable-workflows#how-the-token-works-with-reusable-workflows dropping this here for my future reference.

**Comment by woodruffw** [comment]
Another possibly useful cross-reference: https://github.com/pypi/warehouse/issues/11096

**Comment by segiddins** [comment]
@ybiquitous there's some work we'd have to do here to support re-usable workflows outside of the calling repository.

Likely this will involve an extra (optional) input field for the reusable workflow, and the existing field will continue to refer to the calling (top-level) workflow (and be the `job_workflow_ref` as well if the reusable workflow is not specified).

**Comment by ybiquitous** [comment]
> Likely this will involve an extra (optional) input field for the reusable workflow, and the existing field will continue to refer to the calling (top-level) workflow

I'm okay if reusable workflows are supported. 👌🏼

**Comment by ronaldtse** [comment]
Just hit the same problem with setting up the trusted publisher flow from a reusable workflow as well.

**Comment by jenshenny** [comment]
https://github.com/rubygems/rubygems.org/pull/6184 was just merged adding support. Folks can read the [trusted publishing](https://guides.rubygems.org/trusted-publishing/adding-a-publisher/) docs for more detail

**Comment by ronaldtse** [comment]
Thank you for the awesome news @jenshenny !

**Comment by tvdeyen** [comment]
Unfortunately it does not work (at least [not for us](https://github.com/AlchemyCMS/alchemy-devise/actions/runs/21598480673/job/62237508965)). Following the documentation and using the official `rubygems/release-gem@v1` GH action. The error message is

```
No trusted publisher configured for this workflow found on https://rubygems.org/ for audience rubygems.org
```

This is the workflow in the gem

https://github.com/AlchemyCMS/alchemy-devise/blob/main/.github/workflows/release.yml#L14

This is the shared workflow

https://github.com/AlchemyCMS/.github/blob/main/.github/workflows/release.yml#L51

all following the official documentation and configured as documented

<img width="1926" height="1284" alt="Image" src="https://github.com/user-attachments/assets/ebad9364-53b7-494d-9bb6-f02bbfdd28c1" />

Any ideas what might be wrong?

**Comment by ybiquitous** [comment]
The `No trusted publisher configured` problem looks like the same issue that I opened last week:
- https://github.com/rubygems/configure-rubygems-credentials/issues/349

**Comment by tvdeyen** [comment]
@jenshenny sorry for tagging, but can we get some attention on this? 🙏🏻

**Comment by jenshenny** [comment]
@tvdeyen apologies for dropping the ball on this, we're actively looking into it. Since https://github.com/rubygems/configure-rubygems-credentials/issues/349 is open and has more discussion, we'll be posting progress there.

**Comment by tvdeyen** [comment]
thanks @jenshenny added comments on the issue 2 months ago, but haven't heared anything since. is this actively worked on?
